acceptance/tests/restart_host_run_puppet.rb

```diff
@@ -48,7 +48,16 @@
 
   step "restart each agent" do
     applicable_agents.each do |agent|
-      agent.reboot
+      # Work-around reboot hanging on reply (BKR-971)
+      rebooter = Thread.start { agent.reboot }
+      sleep 5
+      begin
+        agent.close
+        rebooter.join
```

Comment:
If `close` raises, we don't `join`. Should we move `join` after the begin/rescue?